src/metabase/models/collection/graph.clj

```diff
@@ -85,10 +85,11 @@
   ([collection-ids collection-namespace]
    (let [group-id->perms (group-id->permissions-set)]
      {:revision (c-perm-revision/latest-id)
-      :groups   (into {} (for [group-id (t2/select-pks-set PermissionsGroup)]
-                           {group-id (group-permissions-graph collection-namespace
-                                                              (group-id->perms group-id)
-                                                              collection-ids)}))})))
+      :groups   (pmap
```

Comment:
Switching this to pmap, without doall, since it will be consumed by the response value, hence realized.